android/KMEA/app/src/main/assets/android-host.js

```diff
@@ -235,7 +235,7 @@ function updateKMText(text) {
       text = '';
   }
 
-  console_debug('updateKMText(text='+text+') context.value='+keyman.context.getText());
+  console_debug('updateKMText(text=' + text + ') with: \n' + build_context_string(keyman.context));
```

Comment:
Maybe just me, but these console_debugs feel more intuitive at the end of the functions (updateKMText and updateKMSelectionRange) where I can visually compare the app context with what's in the console log

Reply:
I was just working with them where they were.  I'm not opposed to having both a "before" and an "after".